sodiumoxide/sodiumoxide.rs

```diff
@@ -41,9 +41,18 @@ mod aead {
                 b.bytes = $input_len;
                 let key = secretbox::gen_key();
                 let nonce = secretbox::gen_nonce();
-                let plaintext = [0u8; $input_len];
+                let mut plaintext = vec![0u8; $input_len];
                 b.iter(|| {
-                    seal(&plaintext, &nonce, &key)
+                    let out = seal(&plaintext, &nonce, &key);
+                    // XXX: secretbox doesn't support in place encryption, so
+                    // fake it until we change this to use the aead encrypt
+                    // function directly.
+                    // out is a bit larger than plaintext. This ignores the
+                    // bytes in the mismatch.
```

Comment:
First, thanks for the update! This is looking better.

But, the example code I gave you is bad and doesn't actually match how the *ring* benchmark works. The *ring* benchmark writes *all* the output bytes to the output. That means that the `plaintext` buffer needs to be large enough to contain the full ciphertext; i.e. the loop needs to be `0..out.len()`.

NIT: For clarify, I suggest that you don't use the name "plaintext" since it will be ciphertext on the way out.